services/src/main/java/org/keycloak/services/resources/KeycloakApplication.java

```diff
@@ -107,6 +109,7 @@ public KeycloakApplication() {
             singletons.add(new RealmsResource());
             singletons.add(new AdminRoot());
             classes.add(ThemeResource.class);
+            classes.add(CachingGZIPEncodingInterceptor.class);
```

Comment:
Should this be a singleton instead?

Reply:
it should be added to the providers, I thought that was what the `classes` where for...

Reply:
classes.add - adds providers where there is a new instance created for each request, while singletons.add adds providers where the same instance is used for all requests. If this is changed to singletons.add then you can have some code in the constructor that is only ran once and not for every request + there's less classes for gc.

Reply:
I don't know why, but when I add it to `singletons` it doesn't use the Interceptor.